.pre-commit-config.yaml

```diff
@@ -80,6 +80,12 @@ repos:
       hooks:
           - id: rapids-dependency-file-generator
             args: ["--clean"]
+    - repo: https://github.com/shellcheck-py/shellcheck-py
+      rev: v0.10.0.1
+      hooks:
+        - id: shellcheck
+          args: ["--severity=warning"]
+          files: ^ci/
```

Comment:
Why not run this on all shell scripts in the repo?

I especially would love to have this coverage on `build.sh` (at the repo root), as that's a public interface into building the components of this project. It's be nice to have `shellcheck` catch issues in changes there.

Reply:
Yeah, `build.sh` coverage is definitely on my list -- I wanted to start with `ci` scripts to get `shellcheck` in place, then we can start to expand coverage to other shell scripts in the repository.

Reply:
ok just wanted to be sure I understood the reasoning. As long as we'll come back again and do the rest, that's fine.

Reply:
I noticed that in the new state of this PR, there are now changing in other files outside of `ci/`... did you manually run `shellcheck` on those? If so, does that mean this filter could now be removed?

Reply:
It wouldn't pass yet as I haven't cleaned up all of `build.sh`, but that's on my list.

I mostly wanted to make sure we actually had the `VALIDARGS` check in `build.sh`, since otherwise there's no point in defining `VALIDARGS`